src/backend/distributed/commands/create_citus_local_table.c

```diff
@@ -0,0 +1,911 @@
+#include "distributed/listutils.h"
+#include "distributed/metadata_sync.h"
+#include "distributed/multi_partitioning_utils.h"
+#include "distributed/namespace_utils.h"
+#include "distributed/reference_table_utils.h"
+#include "distributed/worker_protocol.h"
+#include "distributed/worker_shard_visibility.h"
+#include "utils/builtins.h"
+#include "utils/fmgroids.h"
+#include "utils/lsyscache.h"
+#include "utils/ruleutils.h"
+#include "utils/syscache.h"
+
+
+static void ErrorIfUnsupportedCreateCitusLocalTable(Relation relation);
+static void ErrorIfUnsupportedCitusLocalTableKind(Oid relationId);
+static uint64 ConvertLocalTableToShard(Oid relationId);
+static void RenameRelationToShardRelation(Oid shellRelationId, uint64 shardId);
+static void RenameShardRelationConstraints(Oid shardRelationId, uint64 shardId);
+static List * GetConstraintNameList(Oid relationId);
+static char * GetRenameShardConstraintCommand(Oid relationId, char *constraintName,
+											  uint64 shardId);
+static void RenameShardRelationIndexes(Oid shardRelationId, uint64 shardId);
+static char * GetDropTriggerCommand(Oid relationId, char *triggerName);
+static char * GetRenameShardIndexCommand(char *indexName, uint64 shardId);
+static void RenameShardRelationNonTruncateTriggers(Oid shardRelationId, uint64 shardId);
+static char * GetRenameShardTriggerCommand(Oid shardRelationId, char *triggerName,
+										   uint64 shardId);
+static void DropRelationTruncateTriggers(Oid relationId);
+static char * GetDropTriggerCommand(Oid relationId, char *triggerName);
+static List * GetExplicitIndexNameList(Oid relationId);
+static void CreateCitusLocalTable(Oid relationId);
+static void FinalizeCitusLocalTableCreation(Oid relationId);
+static List * GetShellTableDDLEventsForCitusLocalTable(Oid relationId);
+static void ExecuteAndLogDDLCommandList(List *ddlCommandList);
+static void DropAndMoveDefaultSequenceOwnerships(Oid sourceRelationId,
+												 Oid targetRelationId);
+static void ExtractColumnsOwningSequences(Oid relationId, List **columnNameList,
+										  List **ownedSequenceIdList);
+static void DropDefaultColumnDefinition(Oid relationId, char *columnName);
+static List * GetSequencesOwnedByColumn(Oid relationId, AttrNumber attributeNumber);
+static void TransferSequenceOwnership(Oid ownedSequenceId, Oid targetRelationId,
+									  char *columnName);
+static void ExecuteAndLogDDLCommand(const char *commandString);
+static void InsertMetadataForCitusLocalTable(Oid citusLocalTableId, uint64 shardId);
+
+
+PG_FUNCTION_INFO_V1(create_citus_local_table);
+
+
+/*
+ * create_citus_local_table creates a citus table from the table with relationId.
+ * by executing the internal method CreateCitusLocalTable.
+ * (See CreateCitusLocalTable function's comment.)
+ */
+Datum
+create_citus_local_table(PG_FUNCTION_ARGS)
+{
+	CheckCitusVersion(ERROR);
+
+	Oid relationId = PG_GETARG_OID(0);
+
+	CreateCitusLocalTable(relationId);
+
+	PG_RETURN_VOID();
+}
+
```

Comment:
Minor:

It could be nice to order the functions in a way that they are called, for example you could place `CreateCitusLocalTable` below this function for better readability